src/lib/ui/Icons/SourceBadge.tsx

```diff
@@ -0,0 +1,62 @@
+import { Tooltip } from "@mantine/core";
+import {
+  IconBrandGoogle,
+  IconFileTypeCsv,
+  IconTable,
+} from "@tabler/icons-react";
+import { match } from "ts-pattern";
+import { DatasetSourceType } from "@/models/datasets/Dataset";
+
+export function SourceBadge({
```

Comment:
This file should go under `src/components/common` - it includes business logic (reference to our Dataset types), therefore it is not a good candidate to have under `src/lib/ui`

Also, a "Badge" is its own type of UI element, so it should not be under an "Icons" directory anyways. Let's just put this under `src/components/common/SourceBadge.tsx`